### PR TITLE
feat: add guild event logging

### LINKED ISSUE #24: Creating a guild-logging system
The logs system should basically send the events that fire to Discord, but just more user-friendly(no guild and such). It will act like the very common logging systems that are made, Event fires > sends the change(change-related events) to the user's preferred channel > and then done.

The events that we should watch fire should be:
- `on_message_edit`
- `on_message_delete`
- `on_guild_channel_delete`
- `on_guild_channel_create`
- `on_guild_channel_update`
- `on_thread_create`
- `on_thread_update`
- `on_thread_delete`
- `on_guild_emojis_update`
- `on_invite_create`
- `on_invite_delete`
- `on_member_join`
- `on_member_remove`
- `on_guild_stickers_update`
- `on_guild_role_create`
- `on_guild_role_delete`
- `on_guild_role_update`
- `on_bulk_message_delete`

These preferebly should use `on_raw_*` events, due to caching, however if there's another workaround, that'd be appreciated.

### PR DESCRIPTION
**Pull request by ooliver1**
closes #24

## Reasons For Seemingly Unrelated Changes

`ezar/__main__.py` - embedded modules such as `ezar.cogs.logs.messages` can be resolved - better directory structure
`ezar/helpers/components.py` - so slash command interaction's responses can be edited, instead of component interactions' messages being edited

## Examples

message edit
message delete
bulk message delete
channel create
channel edit
cahnnel delete
thread create
thread edit
thread delete
emoji create (update event)
emoji delete (update event)
sticker create (update event)
sticker delete (update event)
invite create
invite delete
member join
member leave
role create
role update
role deleted

### CODE REVIEW

File: ezar/cogs/logs/messages.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            unique_users = ", ".join(
+                f"`{user}`" for user in set(str(m.author) for m in messages)
+            )
+            description = (
+                f"{len(messages)} messages in {first_message.channel.mention} "
+                f"(`{first_message.channel.id}` - `{first_message.channel.name}`) by"
+                f"{unique_users} were deleted"
+            )
+
+            embed = Embeb(title="Multiple Messages Deleted", description=description)
```
Comment (toolifelesstocode) [inline ezar/cogs/logs/messages.py:112]: Could we have it send a .txt file to make it show the deleted messages? Yes it does state that multiple messages were deleted, but it'd be great if it showed what messages were deleted, if possible.

File Structure:
```txt
Messages Deleted: {count}

{Message} - {Message author}
``` 
or something like that

**Review by toolifelesstocode — CHANGES_REQUESTED** [review]
Looks awesome! Just have a few requests :)

**Review by toolifelesstocode — APPROVED** [review]
lgtm

**Comment by ooliver1** [comment]
LETS FUCKING GOOOO

**Pull request merged**